### PR TITLE
bpo-41923: PEP 613: Add TypeAlias to typing module

### PR DESCRIPTION
**Pull request by east825**
This special marker annotation is intended to help in distinguishing
proper PEP 484-compliant type aliases from regular top-level variable 
assignments.

https://bugs.python.org/issue41923

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2020-10-03-23-14-50.bpo-41923.Buonw9.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Implement :pep:`613`, introducing :class:`typing.TypeAlias` annotation.
```
Comment (gvanrossum) [inline Misc/NEWS.d/next/Library/2020-10-03-23-14-50.bpo-41923.Buonw9.rst]: Could you mention that you just copied this code from typing_extensions? It doesn't seem fair to claim authorship for this code, as you imply here and in your change to Misc/ACKS. You only deserve credit for figuring out where to put the new code. :-)

Comment (east825) [reply Misc/NEWS.d/next/Library/2020-10-03-23-14-50.bpo-41923.Buonw9.rst]: Sure, I could, but I've authored it in typing_extensions :)

**Comment by east825** [comment]
I can try to come up with proper documentation for `TypeAlias` in this PR or separately, especially if someone agrees to proofread me.

**Comment by gvanrossum** [comment]
I’d be happy to proofread!--
--Guido (mobile)

**Review by east825 — COMMENTED** [review]
Can I consider the problem with mentioning me in ACKS resolved, since I only ported [my own PR in typing_extensions](https://github.com/python/typing/pull/732)? I wouldn't ever have claimed someone else's authorship(

**Comment by east825** [comment]
I've added minimal documentation, but already not sure about a couple of things:
* Should all the examples of type aliases, first of all, those in the documentation for typing module, be updated to use TypeAlias?
* I don't know how to better put, both in What's New and in the documentation for typing, that people are encouraged to use the new annotation, however, the former "implicit" syntax will continue to be supported by type checkers. PEP 613 doesn't give any deprecation timeline in this regard.

File: Doc/whatsnew/3.10.rst [hunk 2]
```diff
@@ -82,10 +82,50 @@ New Features
 * :pep:`618`: The :func:`zip` function now has an optional ``strict`` flag, used
   to require that all the iterables have an equal length.
 
-* :pep:`613`: The :mod:`typing` module now has a special annotation
-  :class:`TypeAlias` to explicitly declare :pep:`484`-compliant type aliases,
-  better differentiating them from regular assignments.
+PEP613: TypeAlias Annotation
+----------------------------
```
Comment (gvanrossum) [inline Doc/whatsnew/3.10.rst]:
```suggestion
PEP 613: TypeAlias Annotation
-----------------------------
```

File: Doc/whatsnew/3.10.rst [hunk 3]
```diff
@@ -82,10 +82,50 @@ New Features
...
+
+   StrCache: TypeAlias = 'Cache[str]'  # a type alias
+   LOG_PREFIX = 'LOG[DEBUG]'  # a module constant
+
+See :pep:`613` for more details.
+
+(Contributed by Mikhail Golubev in :issue:`41923`.)
+
+PEP604: New Type Operator
+-------------------------
```
Comment (gvanrossum) [inline Doc/whatsnew/3.10.rst]:
```suggestion
PEP 604: New Type Union Operator
--------------------------------
```

File: Doc/whatsnew/3.10.rst [hunk 4]
```diff
@@ -82,10 +82,50 @@ New Features
...
+using :data:`typing.Union`, especially in type hints (annotations).
+
+In previous versions of Python, to apply a type hint for functions accepting
+arguments of multiple types, :data:`typing.Union` was used::
+
+   def square(number: Union[int, float]) -> Union[int, float]:
+       return number ** 2
+
+
+Now, type hints can be written in a more succinct manner::
```
Comment (gvanrossum) [inline Doc/whatsnew/3.10.rst]:
```suggestion
Type hints can now be written in a more succinct manner::
```

File: Misc/NEWS.d/next/Library/2020-10-03-23-14-50.bpo-41923.Buonw9.rst (hunk 1 above)
Comment (gvanrossum) [reply Misc/NEWS.d/next/Library/2020-10-03-23-14-50.bpo-41923.Buonw9.rst]: Touché!

**Review by gvanrossum — APPROVED** [review]
I will apply my text nits and land. Thanks for your contribution, and welcome to the walhalla named Misc/ACKS. :-)

**Comment by gvanrossum** [comment]
> I've added minimal documentation, but already not sure about a couple of things:
> 
>     * Should all the examples of type aliases, first of all, those in the documentation for typing module, be updated to use TypeAlias?

I don't think so. Let's wait for this to be implemented in all static checkers before we go down that road.

>     * I don't know how to better put, both in What's New and in the documentation for typing, that people are encouraged to use the new annotation, however, the former "implicit" syntax will continue to be supported by type checkers. PEP 613 doesn't give any deprecation timeline in this regard.

I don't think deprecation is needed at this point -- some type checkers may start enforcing it but they will communicate to their own users.

File: Doc/whatsnew/3.10.rst [hunk 5]
```diff
@@ -82,6 +82,27 @@ New Features
...
+
+   StrCache: TypeAlias = 'Cache[str]'  # a type alias
+   LOG_PREFIX = 'LOG[DEBUG]'  # a module constant
+
+See :pep:`613` for more details.
+
+(Contributed by Mikhail Golubev in :issue:`41923`.)
+
 PEP604: New Type Operator
 -------------------------
```
Comment (gvanrossum) [inline Doc/whatsnew/3.10.rst]:
```suggestion
PEP604: New Type Union Operator
-------------------------------
```

**Comment by gvanrossum** [comment]
Thanks again!

**Comment by east825** [comment]
> I will apply my text nits and land. Thanks for your contribution, and welcome to the walhalla named Misc/ACKS. :-)

Thanks for admitting me there!

**Pull request merged**